doc/cprover-manual/loop-invariant-synthesizer.md

```diff
@@ -0,0 +1,28 @@
+[CPROVER Manual TOC](../)
+
+## Loop Invariant Synthesis with goto-instrument
+
+This is a loop invariant synthesizer built in goto-instrument that 
+synthesize [invariant clause](contracts-invariants.md) for a 
+pre-existing GOTO binary, annotate the invariant clause into 
+the GOTO binary, and produce the annotated binary as output.
+
+Loop invariant synthesized by the synthesizer must be inductive 
+in the sense that 
+1. the instrumented loop invariant checks success, no matter other 
+assertions fail or success;
+2. the base-case is from the whole GOTO program with a main 
+function as the entry point.
+
+For different synthesis strategy, there are different prerequisites
+on the input GOTO binary, and different guarantees on the synthesized
+loop invariants, summarized in the
+following table. 
+
+Strategy         | Flag                                           | Prerequisites                                      | Guarantees
+-----------------|------------------------------------------------|----------------------------------------------------|----
+Enumerative      | --synthesize-loop-invariants                   | The input GOTO binary contains only pointer checks | All checks success in the annotated GOTO binary
```

Comment:
I am unclear on the restrictions part here. When you say "contains only pointer checks," are you excluding all arithmetic checks (overflows, conversions etc.)? Does pointer checks include all kinds of pointer checks (overflows, primitive checks etc.) or just pointer validity checks?

Should they not appear anywhere in the goto file, or just the loop body? What is the expected behavior when they do appear, do we error out?

Reply:
I refer by _pointer checks_ the checks added by goto-instrument with the flag ```--pointer-check```. I will clarify it.

When there checks are than pointer checks, there are two possibility. 
1. Those checks don't depend on any loop invariant. The synthesizer will work as those checks don't exist. 
2. Those checks depend on some loop invariant. We will warm the user that there exists non-pointer-checks and the enumeration is likely not to terminate.

Reply:
I see, thanks. Let's add this behavior to the document

Reply:
To be more accurate, I removed the restriction part. Instead we say the synthesizer is _designed_ for some kind of GOTO binaries---it works best for them, but still have a chance to solve problems with other checks.